### PR TITLE
cgns 3.3

### LINKED ISSUE #3867: CGNS 3.3
### Update CGNS to version 3.3?

CGNS has been updated in January, and I for one use it occasionally. :)

This update might pose some problems though, as with no additional configuration settings, the cg_sizet type is changed from `int` to `unsigned long` (I think), so some code using this library may no longer compile if it assumes `int` and `cg_sizet` are the same type.

Also, the names of configuration options seems to have changed. I successfully build 3.3 with the following formula. Changes made include:
- Updated homepage
- Making fortran bindings optional
- Updating the names of the defines / configuration parameters for cmake

```ruby
class Cgns < Formula
  homepage "http://cgns.org/"
  url "https://github.com/CGNS/CGNS/archive/v3.3.0.tar.gz"
  sha256 "8422c67994f8dc6a2f201523a14f6c7d7e16313bdd404c460c16079dbeafc662"
  revision 1

  depends_on :fortran => :optional
  depends_on "cmake" => :build
  depends_on "hdf5" => :recommended
  depends_on "szip"

  def install
    args = std_cmake_args + [
      "-DCGNS_ENABLE_TESTS=YES",
    ]

    args << "-DCGNS_ENABLE_64BIT=YES" if Hardware.is_64_bit? && MacOS.version >= :snow_leopard
    args << '-DCGNS_ENABLE_FORTRAN=YES' if build.with? "fortran"

    if build.with? "hdf5"
      args << "-DCGNS_ENABLE_HDF5=YES"
      args << "-DHDF5_NEED_ZLIB=YES"
      args << "-DHDF5_NEED_SLIB=YES"
      args << "-DCMAKE_SHARED_LINKER_FLAGS=-lhdf5"
    end

    mkdir "build" do
      system "cmake", "..", *args
      system "make"
      system "ctest --output-on-failure"
      system "make", "install"
    end
  end
end
```

### PR DESCRIPTION
**Pull request by q-p**
### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### Updates to existing formula: have you

- [x] Removed the `revision` line, if any, when bumping the version number?
- [x] Not altered the `bottle` section?
- [ ] Checked that the tests still pass?

Closes #3867.

### CODE REVIEW

File: cgns.rb [hunk 1]
```diff
@@ -1,40 +1,63 @@
...
+          return 1; // should fail!
+        printf(\"%d.%d.%d\\n\",CGNS_VERSION/1000,(CGNS_VERSION/100)%10,(CGNS_VERSION/10)%10);
+        return 0;
+      }
+    EOS
+    if build.with? "hdf5"
+      compiler = "h5cc"
+    else
+      compiler = ENV.cc
+    end
```
Comment (dpo) [inline cgns.rb]: You can't really use `build.with?` in `test`. Instead you need to look at `Tab.for_formula["cgns"].with? "hdf5"`. So you could use the one liner
```
compiler = Tab.for_formula["cgns"].with?("hdf5") ? "h5cc" : ENV.cc
```

**Comment by dpo** [comment]
Would you mind adding paraview to your pull request and bump its revision number? It takes a long time to build and will time out. I'll restart the build at a time where I'm unlikely to bother many people.

**Comment by q-p** [comment]
Part of the problem seems to be that the original version of the formula was using the old (but even incorrect for cgns 3.2) configuration names, and as such never built with HDF5 (or the Fortran interface for that matter)...

**Comment by q-p** [comment]
At least some of the paraview errors are due to it being incompatible with the current version of ffmpeg.

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by dpo** [comment]
Sorry for the long delay. Could you please rebase? I just updated `paraview` to fix the FFMPEG error. Please bump it another time in this PR.

**Comment by dpo** [comment]
@ilovezfs the brew bot seems to be stuck on this one and on #4032. Do you know how to unstick it?

**Comment by ilovezfs** [comment]
looks like it's fixed now

**Comment by dpo** [comment]
Still seems stuck.

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Right because it was aborted so the status was never reported back ... https://bot.brew.sh/job/Homebrew%20Science%20Pull%20Requests/5566/

The next pass or fail will update the status, whether triggered by @BrewTestBot or just clicking the rebuild button.

**Comment by dpo** [comment]
Finally!

**Comment by dpo** [comment]
@q-p Thanks for your patience and contribution!

**Pull request closed without merging**